xbmc/addons/gui/GUIDialogAddonSettings.cpp

```diff
@@ -40,12 +47,49 @@ bool CGUIDialogAddonSettings::OnMessage(CGUIMessage& message)
   {
     case GUI_MSG_CLICKED:
     {
-      if (message.GetSenderId() == CONTROL_SETTINGS_CUSTOM_BUTTON)
+      const auto id = message.GetSenderId();
```

Comment:
I would not use `auto` for trivial types with short names. `int`in this case.

Reply:
Think also about integers makes a `const` sense, maybe the compiler makes it a bit better as it see that no edit comes. Also one dev has brought something about, maybe @wsnipex not sure?

Reply:
yes, use `const` whenever possible inside method implementation. MAkes the code more readable and pretty clear, that you do not intend to modify the variqble afetr first assignment. Also, helps the compielr to optimize code. See also: http://www.gotw.ca/publications/advice98.htm